### PR TITLE
Add replace file contents functionality

### LINKED ISSUE #32: Error when beautifier is installed without unibeautify peer dependency
Steps to reproduce:

1. Globally install an @unibeautify beautifier
2. Ensure that unibeautify is not installed globally
3. Run the cli, e.g. `unibeautify list -l`

Result:  It throws an error:

```
(node:7924) UnhandledPromiseRejectionWarning: NestedError: Could not require module '@unibeautify/beautifier-eslint'
    at requireg (/Users/stevenzeck/Desktop/Development/unibeautify/core/unibeautify-cli/node_modules/requireg/lib/requireg.js:14:11)
    at beautifierNames.map.beautifierName (/Users/stevenzeck/Desktop/Development/unibeautify/core/unibeautify-cli/dist/index.js:27:16)
    at Array.map (<anonymous>)
    at loadBeautifiers (/Users/stevenzeck/Desktop/Development/unibeautify/core/unibeautify-cli/dist/index.js:26:41)
    at findInstalledBeautifiers.then (/Users/stevenzeck/Desktop/Development/unibeautify/core/unibeautify-cli/dist/index.js:34:16)
Caused By: Error: Cannot find module 'unibeautify'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/@unibeautify/beautifier-eslint/dist/src/index.js:3:23)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
(node:7924) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7924) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### PR DESCRIPTION
**Pull request by szeck87**
* Adds replace functionality (via the `-r` argument)

Closes #32

### CODE REVIEW

**Comment by szeck87** [comment]
I'll need to replace the `fs.promises` functions with the regular ones so it runs in node 8.

**Comment by lassik** [comment]
Good to have this feature.

I'd suggest renaming the flag to `-i` / `--inplace`. There's a fairly strong precedent from Unix tradition for the in-place editing flag to be called `-i` because `sed -i` has been standard since forever. Likewise, there's a very strong precedent for `-r` to mean recursive (not sure whether we'll ever need a recursive flag). Another common meaning for `-r` is read or read-only, which is the opposite of what we are doing here.

`clang-format`, `dfmt` and `yq` already use `-i` for in-place. `gofmt` uses `-w` ("write") and `prettier` has `--write` but no `-w`.

I dumped the help messages of lots of formatters into a Gist to help decide option names: https://gist.github.com/lassik/9ec41eaff379ab655516ee69cf42f8e4

**Comment by Glavin001** [comment]
@szeck87 Feel free to change the CLI arguments as you see fit based on @lassik 's suggestions. Thanks everyone!

**Comment by szeck87** [comment]
@Glavin001 any thoughts on code coverage for the remaining lines in BeautifyCommand.ts?  36,97,98,100,136

**Comment by Glavin001** [comment]
@szeck87 Not yet. I think additional coverage can be in a new PR, for these lines.

**Comment by szeck87** [comment]
@Glavin001 this is ready for review.

**Comment by lassik** [comment]
Just thought to check whether Node's `fs.writeFile()` is safe against errors. Unfortunately, it doesn't seem to be. I couldn't find anything [in the docs](https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback) or [in the source code](https://github.com/nodejs/node/blob/master/lib/fs.js#L1139) that preserves the old contents of the file in case there's some kind of I/O error.

Under Unix, standard practice is to write the new contents of `somefile` first into a temp file `somefile.new` and then rename `somefile.new` back to `somefile`. The [Unix `rename()` system call](http://pubs.opengroup.org/onlinepubs/9699919799/functions/rename.html) will atomically replace the old file with the new one. If an error occurs, the OS kernel guarantees that the old file remains intact. Not sure how this is done on Windows normally.

File: src/utils.ts [hunk 1]
```diff
@@ -25,8 +25,13 @@ export function isBeautifierPackageName(packageName: string): boolean {
 
 export function loadBeautifiers(beautifierNames: string[]): Unibeautify {
   const beautifiers = beautifierNames.map(beautifierName => {
-    return requireg(beautifierName).default;
-  });
+    try {
+      return requireg(beautifierName).default;
+    } catch (error) {
+      console.error(error);
+      return null;
```
Comment (Glavin001) [inline src/utils.ts]: @szeck87 Did this work?

File: src/commands/BeautifyCommand.ts [hunk 2]
```diff
@@ -112,6 +113,32 @@ export class BeautifyCommand extends BaseCommand {
       });
     });
   }
+
+  private beautifySuccess({
```
Comment (Glavin001) [inline src/commands/BeautifyCommand.ts]: I do not think `beautifySuccess` describes *what* it does, only *when* to call it.

Something like `writeToFileOrStdout` would be better.

File: src/utils.ts (hunk 1 above)
Comment (szeck87) [reply src/utils.ts]: Yes it did, no more unhandled promise rejections.

File: src/commands/BeautifyCommand.ts (hunk 2 above)
Comment (szeck87) [reply src/commands/BeautifyCommand.ts]: @Glavin001 changed it.

**Comment by lassik** [comment]
Here's an NPM package that provides a safer `fs.writeFile()` as outlined above: https://www.npmjs.com/package/write-file-atomic But it seems fairly complicated, and as far as I can tell there is no Windows support.

**Comment by Glavin001** [comment]
@lassik separate PRs are very welcome for safer write support.

@szeck87 for now, let's merge this in.

**Pull request merged**